osu.Server.Queues.ScoreStatisticsProcessor/Processors/PerformanceProcessor.cs

```diff
@@ -111,6 +105,55 @@ public static bool AllModsValidForPerformance(Mod[] mods)
             return true;
         }
 
+        private static DifficultyAttributes? queryAttributes(SoloScoreInfo score)
+        {
+            var req = new WebRequest(AppSettings.DIFFCALC_ENDPOINT)
+            {
+                Method = HttpMethod.Post,
+                ContentType = "application/json",
+                Timeout = int.MaxValue,
+                AllowInsecureRequests = true
+            };
+
+            req.AddRaw(JsonConvert.SerializeObject(new DifficultyAttributesRequestData
+            {
+                BeatmapId = score.beatmap_id,
+                RulesetId = score.ruleset_id,
+                Mods = score.mods
+            }));
+
+            DifficultyAttributes? attributes = null;
+
+            req.Finished += () =>
+            {
+                if (req.ResponseStream == null)
```

Comment:
Any reason to not just get the string immediately?

```suggestion
                if (!(req.GetResponseString() is string responseJson))
```

to avoid the null suppression below?